src/interval_sets.jl

```diff
@@ -384,15 +425,15 @@ end
 
 """
     find_intersections(
-        x::Union{AbstractInterval, IntervalSet},
-        y::Union{AbstractInterval, IntervalSet},
+        x::AbstractVector{<:AbstractInterval},
+        y::AbstractVector{<:AbstractInterval}
     )
 
 Returns a `Vector{Vector{Int}}` where the value at index `i` gives the indices to all
 intervals in `y` that intersect with `x[i]`.
 """
-function find_intersections(x_::AbstractIntervals, y_::AbstractIntervals)
-    xa, ya = IntervalSet(x_), IntervalSet(y_)
+function find_intersections(x_::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, y_::Union{AbstractInterval, AbstractVector{<:AbstractInterval}})
+    xa, ya = vcat(x_), vcat(y_)
```

Comment:
Use multiple dispatch to clean things up here:
```suggestion
find_intersections(x, y) = find_intersections(vcat(x), vcat(y))

function find_intersections(xa::AbstractVector{<:AbstractInterval}}, ya::AbstractVector{<:AbstractInterval}})
```